### ISSUE TITLE
Auto-Complete behavior

### ISSUE THREAD

**Issue by Zhaocl1997** [tusen-ai/naive-ui#4658]
### TuSimple/naive-ui version (版本)
2.34.3

### Vue version (Vue 版本)
3.3.0-alpha.5

### Browser and its version (浏览器及其版本)
Chrome (111.0.5563.147)

### System and its version (系统及其版本)
Windows10

### Node version (Node 版本)
16.19.1

### Reappearance link (重现链接)
https://codesandbox.io/s/agitated-sea-79166k

### Reappearance steps (重现步骤)
1. Input something
2. Click outside
3. Click clear icon to clear the content
4. The select menu show up for a tick and disappear

### Expected results (期望的结果)
After click the clear icon, select menu should not show up anymore

### Actual results (实际的结果)
The select menu flashed for a tick and disappear, I think it's bad user experience

### Remarks (补充说明)
I think it's not a design-so behavior right?
I've looked deeper and find out that it's the n-input default behavior that the input would auto-focus after clicked clear icon (in the n-input `handleMouseDown` function).
Do you guys think it's ok to add a flag for n-input like `focusAfterClear` which is default true, and in auto-complete, just pass `:focus-after-clear="false"` to n-input and the behavior is gone.
I can make a pr, with tested case and document added. Just wondering how you guys think about it.

**Comment by amadeus711** [comment]
![20230330095650](https://user-images.githubusercontent.com/46394163/228708307-3a413ad4-61f8-48cb-bcf9-e0afc4a37574.gif)
Not repeated on Mac. If this situation exists on Windows, it should be fixed, welcome PR. As for the `focus` related proposal mentioned in Remarks, it does not conform to the team's design for the `Input`.

**Comment by Zhaocl1997** [comment]
Sorry I think the step I didn't explain properly, you should input something => click outside => click clear icon, the select menu just flash for a tick

**Comment by jahnli** [comment]
![chrome_y9WHzWCcVz](https://user-images.githubusercontent.com/20943608/229760080-a573c299-f3e2-47f5-a22b-abd18e840853.gif)

**Comment by jahnli** [comment]
@Zhaocl1997  是这个意思 ？

**Comment by Zhaocl1997** [comment]
> @Zhaocl1997 是这个意思 ？

对 就这这样的闪烁 感觉体验不是很好

**Comment by jahnli** [comment]
> > @Zhaocl1997 是这个意思 ？
> 
> 对 就是这样的闪烁 感觉体验不是很好

我去看了 `element-plus` 和其他UI库，貌似都是这种效果，如果作者确定是需要改，我会尝试`PR`

**Comment by 07akioni** [comment]
After some discovery, the fix in #5064 has pitfalls that make other components misbehave.

The behavior that the issue mentions is expected but wired in real life. I provide `show-empty` prop and `empty` slot on `n-auto-complete` to make sure it can have reasonable behavior under some configurations.

```html
<n-auto-complete
  :get-show="() => true"
  show-empty
>
  <template #empty>
    <n-empty :show-icon="false">请输入</n-empty>
  </template>
</n-auto-complete>
```
